modules/hyprlock/hm.nix

```
@@ -1,21 +1,29 @@
 { config, lib, ... }:
 
 with config.lib.stylix;
+
+let
+  cfg = config.stylix.targets.hyprlock;
+in
 {
-  options.stylix.targets.hyprlock.enable = mkEnableTarget "Hyprlock" true;
+  options.stylix.targets.hyprlock = {
+    enable = mkEnableTarget "Hyprlock" true;
+    useWallpaper = mkEnableWallpaper "Hyprlock" true;
+  };
 
-  config =
-    lib.mkIf (config.stylix.enable && config.stylix.targets.hyprlock.enable)
-      {
-        programs.hyprlock.settings = {
-          background.path = "${config.stylix.image}";
-          input-field = with colors; {
-            outer_color = "rgb(${base03})";
-            inner_color = "rgb(${base00})";
-            font_color = "rgb(${base05})";
-            fail_color = "rgb(${base08})";
-            check_color = "rgb(${base0A})";
-          };
-        };
+  config = lib.mkIf (config.stylix.enable && cfg.enable) {
+    programs.hyprlock.settings = {
+      background = {
+        path = lib.mkIf cfg.useWallpaper config.stylix.image;
+        color = "rgb(${base00})";
```

A bit late on this one, but I believe this should be:

```suggestion
        color = "rgb(${colors.base00})";
```

---

whoops, i think you're right. what's best practice for me to do here, open a new PR?

---

Yes, just go ahead and open a new PR and we'll get it merged ASAP